src/frontend/src/lib/utils/qr-code.utils.ts

```diff
@@ -1,7 +1,49 @@
-import type { QrResponse, QrStatus } from '$lib/types/qr-code';
+import {
+	urnNumericParams,
+	urnStringParams,
+	type DecodedUrn,
+	type QrResponse,
+	type QrStatus
+} from '$lib/types/qr-code';
 import { decodePayment } from '@dfinity/ledger-icrc';
 import { isNullish, nonNullish, type Token } from '@dfinity/utils';
 
+export const decodeUrn = (urn: string): DecodedUrn | undefined => {
+	const regex = /^([a-zA-Z]+):([a-zA-Z0-9\-.]+)(@(\d+))?(\/([a-zA-Z]+))?(\?(.*))?$/;
+
+	const match = urn.match(regex);
+	if (isNullish(match)) {
+		return undefined;
+	}
+
+	const [_, prefix, destination, , networkId, , functionName, , queryString] = match;
+
+	const params: { [key: string]: string | number | undefined } = {};
+
+	if (queryString) {
+		const queryParams = new URLSearchParams(queryString);
+		queryParams.forEach((value, key) => {
```

Comment:
nitpick: would be nice to map the params in a functional way, that's generally the approach in the dapp

Reply:
changed